x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/AutomataMatchEvaluator.java

```diff
@@ -22,22 +22,25 @@
 import org.elasticsearch.xpack.ql.tree.Source;
 
 /**
- * {@link EvalOperator.ExpressionEvaluator} implementation for {@link RegexMatch}.
+ * {@link EvalOperator.ExpressionEvaluator} implementation for {@link AutomataMatch}.
  * This class is generated. Do not edit it.
  */
-public final class RegexMatchEvaluator implements EvalOperator.ExpressionEvaluator {
+public final class AutomataMatchEvaluator implements EvalOperator.ExpressionEvaluator {
   private final Warnings warnings;
 
   private final EvalOperator.ExpressionEvaluator input;
 
-  private final CharacterRunAutomaton pattern;
+  private final ByteRunAutomaton automaton;
+
+  private final String pattern;
 
   private final DriverContext driverContext;
 
-  public RegexMatchEvaluator(Source source, EvalOperator.ExpressionEvaluator input,
-      CharacterRunAutomaton pattern, DriverContext driverContext) {
+  public AutomataMatchEvaluator(Source source, EvalOperator.ExpressionEvaluator input,
+      ByteRunAutomaton automaton, String pattern, DriverContext driverContext) {
```

Comment:
👍 when I implemented it, the performance implications of using `CharacterRunAutomaton` rather than `ByteRunAutomaton` were not clear not me.